src/inference/include/openvino/runtime/intel_gpu/params.hpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header for advanced hardware related properties for GPU plugin
+ *        To use in SetConfig() method of plugins
+ *
+ * @file params.hpp
+ */
+#pragma once
+
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+using gpu_handle_param = void*;
+
+/**
+ * @brief Enum to define context type
+ */
+enum class ContextType {
+    OCL = 0,
+    VA_SHARED = 1,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const ContextType& context_type) {
+    switch (context_type) {
+    case ContextType::OCL:
+        return os << "OCL";
+    case ContextType::VA_SHARED:
+        return os << "VA_SHARED";
+    default:
+        IE_THROW() << "Unsupported context type";
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, ContextType& context_type) {
+    std::string str;
+    is >> str;
+    if (str == "OCL") {
+        context_type = ContextType::OCL;
+    } else if (str == "VA_SHARED") {
+        context_type = ContextType::VA_SHARED;
+    } else {
+        IE_THROW() << "Unsupported context type: " + str;
```

Comment:
IE_THROW => OPENVINO_ASSERT

Reply:
could you please also fix in https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/intel_gpu/ocl/ocl.hpp#L198

Reply:
`OPENVINO_ASSERT` usage would be kind of weird here, as it's supposed to check some statement, but we just need to throw an exception.
Doesn't it make sense to add macro like `OPENVINO_THROW`/`OV_THROW`?

Reply:
@ilyachur could you please advice about it?
as far as I remember, we have `OPENVINO_UNREACHABLE`